rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SendMessageDirectives.scala

```diff
@@ -30,66 +29,30 @@ trait SendMessageDirectives {
+            complete(SendMessageResponse(messageAttributeDigest, digest, messageSystemAttributeDigest, message.id.id, message.sequenceNumber))
         }
       }
     }
   }
 
-  def getMessageAttributes(parameters: Map[String, String]): Map[String, MessageAttribute] = {
-    // Determine number of attributes -- there are likely ways to improve this
-    val numAttributes = parameters
-      .map { case (k, _) =>
-        if (k.startsWith("MessageAttribute.")) {
-          k.split("\\.")(1).toInt
-        } else {
-          0
-        }
-      }
-      .toList
-      .union(List(0))
-      .max // even if nothing, return 0
-
-    (1 to numAttributes).map { i =>
-      val name = parameters("MessageAttribute." + i + ".Name")
-      val dataType = parameters("MessageAttribute." + i + ".Value.DataType")
-
-      val primaryDataType = dataType.split('.')(0)
-      val customDataType = if (dataType.contains('.')) {
-        Some(dataType.substring(dataType.indexOf('.') + 1))
-      } else {
-        None
-      }
-
-      val value: MessageAttribute = primaryDataType match {
-        case "String" =>
-          val strValue = parameters("MessageAttribute." + i + ".Value.StringValue")
-          StringMessageAttribute(strValue, customDataType)
-        case "Number" =>
-          val strValue = parameters("MessageAttribute." + i + ".Value.StringValue")
-          NumberMessageAttribute(strValue, customDataType)
-        case "Binary" =>
-          BinaryMessageAttribute.fromBase64(parameters("MessageAttribute." + i + ".Value.BinaryValue"), customDataType)
-        case "" =>
-          throw new SQSException(s"Attribute '$name' must contain a non-empty attribute type")
-        case _ =>
-          throw new Exception("Currently only handles String, Number and Binary typed attributes")
-      }
-
-      (name, value)
-    }.toMap
-  }
+  def getMessageAttributes(prefix: String)(parameters: Map[String, String]): Map[String, MessageAttribute] = {
+    val messageAttributeNamePattern = s"""$prefix\\.(\\d+)\\.Name""".r
 
-  def getMessageSystemAttributes(parameters: Map[String, String]): Map[String, MessageAttribute] = {
     parameters.flatMap {
-      case (messageSystemAttributeNamePattern(index), parameterName) =>
-        val parameterDataType = parameters(s"MessageSystemAttribute.$index.Value.DataType")
+      case (messageAttributeNamePattern(index), parameterName) =>
+        val parameterDataType = parameters(s"$prefix.$index.Value.DataType")
+
         val parameterValue = parameterDataType match {
-          case "String" => StringMessageAttribute(parameters(s"MessageSystemAttribute.$index.Value.StringValue"))
-          case "Number" => NumberMessageAttribute(parameters(s"MessageSystemAttribute.$index.Value.StringValue"))
+          case "String" => StringMessageAttribute(parameters(s"$prefix.$index.Value.StringValue"))
+          case "Number" => NumberMessageAttribute(parameters(s"$prefix.$index.Value.StringValue"))
```

Comment:
I know it's been like that in the old impl, but shouldn't this be different than `StringValue`?

Reply:
No:

> Amazon SQS supports the following logical data types: String, Number, and Binary. For the Number data type, you must use StringValue.

Reply:
Interesting design choice ;)